satellite/satellitedb/overlaycache.go

```diff
@@ -102,24 +102,39 @@ func (cache *overlaycache) queryFilteredNodes(ctx context.Context, excluded []st
 	}
 	args = append(args, count)
 
-	rows, err := cache.db.Query(cache.db.Rebind(`SELECT id,
-		type, address, free_bandwidth, free_disk, audit_success_ratio,
+	var rows *sql.Rows
+	if distinctIPs {
+		rows, err = cache.db.Query(cache.db.Rebind(`SELECT id,
+		type, address, last_ip, free_bandwidth, free_disk, audit_success_ratio,
+		uptime_ratio, total_audit_count, audit_success_count, total_uptime_count,
+		uptime_success_count
+		FROM (SELECT id, type, address, last_ip, free_bandwidth, free_disk, audit_success_ratio,
+			uptime_ratio, total_audit_count, audit_success_count, total_uptime_count, uptime_success_count,
+			Row_number() OVER(PARTITION BY last_ip ORDER BY RANDOM()) rn
+			FROM nodes
+			`+safeQuery+safeExcludeNodes+`) n
+		WHERE rn = 1
+		ORDER BY RANDOM()
```

Comment:
I don't think this ORDER BY is necessary since you are already randomizing things in the section where you partition by last ip